nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/test/java/org/apache/nifi/cluster/coordination/http/replication/okhttp/OkHttpReplicationClientTest.java

```diff
@@ -0,0 +1,217 @@
+        final String[] methods = new String[] {"DELETE", "delete", "DeLeTe"};
+
+        final NiFiProperties mockProperties = mockNiFiProperties();
+
+        final OkHttpReplicationClient client = new OkHttpReplicationClient(mockProperties);
+
+        for (final String method: methods) {
+            client.prepareRequest(method, headers, null);
+
+            assertEquals(2, headers.size());
+            assertEquals("0", headers.get("Content-Length"));
+        }
+    }
+
+    @Test
+    void testShouldNotReplaceContentLengthHeaderWhenZeroOrNull() {
+        final String method = "DELETE";
+        final String[] zeroOrNullContentLengths = new String[] {null, "0"};
+
+        final NiFiProperties mockProperties = mockNiFiProperties();
+
+        final OkHttpReplicationClient client = new OkHttpReplicationClient(mockProperties);
+
+        final Map<String, String> headers = new HashMap<>();
+        for (final String contentLength: zeroOrNullContentLengths) {
+            headers.put("Content-Length", contentLength);
+            headers.put("Other-Header", "arbitrary value");
+
+            client.prepareRequest(method, headers, null);
+
+            assertEquals(2, headers.size());
+            assertEquals(contentLength, headers.get("Content-Length"));
+        }
+    }
+
+    @Test
+    void testShouldNotReplaceNonZeroContentLengthHeaderOnOtherMethod() {
+        final Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Length", "123");
+        headers.put("Other-Header", "arbitrary value");
+
+        final NiFiProperties mockProperties = mockNiFiProperties();
+
+        final OkHttpReplicationClient client = new OkHttpReplicationClient(mockProperties);
+
+        final String[] nonDeleteMethods = new String[] {"POST", "PUT", "GET", "HEAD"};
+
+        for (final String method: nonDeleteMethods) {
+            client.prepareRequest(method, headers, null);
+
+            assertEquals(2, headers.size());
+            assertEquals("123", headers.get("Content-Length"));
+        }
+    }
+
+    @Test
+    void testShouldUseKeystorePasswordIfKeyPasswordIsBlank() {
+        final Map<String, String> propsMap = new HashMap() {{
+            put(NiFiProperties.SECURITY_TRUSTSTORE, tlsConfiguration.getTruststorePath());
+            put(NiFiProperties.SECURITY_TRUSTSTORE_TYPE, tlsConfiguration.getTruststoreType().getType());
+            put(NiFiProperties.SECURITY_TRUSTSTORE_PASSWD, tlsConfiguration.getTruststorePassword());
+            put(NiFiProperties.SECURITY_KEYSTORE, tlsConfiguration.getKeystorePath());
+            put(NiFiProperties.SECURITY_KEYSTORE_TYPE, tlsConfiguration.getKeystoreType().getType());
+            put(NiFiProperties.SECURITY_KEYSTORE_PASSWD, tlsConfiguration.getKeystorePassword());
+            put(NiFiProperties.SECURITY_KEY_PASSWD, "");
+            put(NiFiProperties.WEB_HTTPS_HOST, "localhost");
+            put(NiFiProperties.WEB_HTTPS_PORT, "51552");
+        }};
```

Comment:
Although this syntax is convenient, creating an anonymous instance of `HashMap` should be avoided.

Reply:
Will change that, thanks for the feedback